src/DatabaseLibrary/query.py

```diff
@@ -350,72 +348,82 @@ def split_sql_script(
         """
         with open(script_path, encoding="UTF-8") as sql_file:
             logger.info("Splitting script file into statements...")
-            statements_to_execute = []
-            if external_parser:
-                split_statements = sqlparse.split(sql_file.read())
-                for statement in split_statements:
-                    statement_without_comments = sqlparse.format(statement, strip_comments=True)
-                    if statement_without_comments:
-                        statements_to_execute.append(statement_without_comments)
-            else:
-                current_statement = ""
-                inside_statements_group = False
-                proc_start_pattern = re.compile("create( or replace)? (procedure|function){1}( )?")
-                proc_end_pattern = re.compile("end(?!( if;| loop;| case;| while;| repeat;)).*;()?")
-                for line in sql_file:
-                    line = line.strip()
-                    if line.startswith("#") or line.startswith("--") or line == "/":
-                        continue
-
-                    # check if the line matches the creating procedure regexp pattern
-                    if proc_start_pattern.match(line.lower()):
-                        inside_statements_group = True
-                    elif line.lower().startswith("begin"):
-                        inside_statements_group = True
-
-                    # semicolons inside the line? use them to separate statements
-                    # ... but not if they are inside a begin/end block (aka. statements group)
-                    sqlFragments = line.split(";")
-                    # no semicolons
-                    if len(sqlFragments) == 1:
-                        current_statement += line + " "
-                        continue
+            return self.split_sql_string(sql_file.read(), external_parser=external_parser)
+
+    def split_sql_string(self, sql_string: str, external_parser: bool = False):
+        if external_parser:
+            return self._split_statements_using_external_parser(sql_string)
+        else:
+            return self._parse_sql_internally(sql_string.splitlines())
```

Comment:
Finding by my side after commit. If general approach is ok, we may shift split to _parse_sql_internally function